### ISSUE TITLE
Aggregate times by tags

### ISSUE THREAD

**Issue by gantir** [jotaen/klog#5]
The eval feature currently takes only one tag at a time. It will be helpful to list aggregates for all categories at the same time. 
eg: 
``` shell
klog eval --tag=ALL

# Output
work  5h
personal 6h
client 3h
```

**Comment by vladdeSV** [comment]
Possibly filter tags by regex?

**Comment by jotaen** [comment]
Thanks for the idea, @gantir , this is now available in klog v1.0 through `klog tags`. In the same spirit you can do `klog report` to get an overview by calendar days.

(And please note that `eval` is now `total`.)

@vladdeSV what use case would you have in mind for the regex? (Will close this ticket here, but happy to continue discussing here anyways.)

**Comment by vladdeSV** [comment]
Might not have though through the regex. When I wrote it I was more interested in getting time by combining tags. Eg. if I want to see how much of a project I also worked overtime at.

Not really a necessary feature though.

**Comment by gantir** [comment]
Super Awesome.

**Comment by gantir** [comment]
@jotaen , 

On further using this option I realised that the tags functionality at global command level is good. However it will help to have it at the total level too

```bash
klog total --tags
```
On further thinking this is probably a better approach than having a separate tags sub-command.  Do you want me to open another issue for the same?

**Comment by jotaen** [comment]
The reason why I prefer separate subcommands is the compatibility of options. E.g. for the `total` command you can do `--diff`, but that wouldn’t make sense for the tag aggregation (since the should-total works on record level). So while there is some overlap between `total`, `tags` and `report`, they do slightly different things in the end.

Maybe you can elaborate why you would prefer something like `total --tags`, so that I can understand it better?

**Comment by gantir** [comment]
> The reason why I prefer separate subcommands is the compatibility of options. E.g. for the `total` command you can do `--diff`, but that wouldn’t make sense for the tag aggregation (since the should-total works on record level). So while there is some overlap between `total`, `tags` and `report`, they do slightly different things in the end.
> 
> Maybe you can elaborate why you would prefer something like `total --tags`, so that I can understand it better?

@jotaen , I understand the issue you mentioned. Can't think of how to resolve.  One example for report I am looking for.
```shell
                     Total
2021 Feb    Fr  5.  7h45m
                       #personal 4h45m
                       #work 3h
                       #client-1 2h
                       #client-2 2h
                   Su  7.       9h
                       #sports 6h
                       #personal 3h
                   ========
                    +16h45m
```
Another example could be just the totals aggregated by tags but for a specific date range (eg: klog tags --date) giving the below output.

```shell
                     Total
2021 Feb    Fr  5.  7h45m
                       #personal 4h45m
                       #work 3h
                       #client-1 2h
                       #client-2 2h
                   Su  7.       9h
                       #sports 6h
                       #personal 3h
                   ========
                    +16h45m
```

**Comment by jotaen** [comment]
I guess the filter options (`--date`, `--before`, etc.) could be added to `tags` and `report` as well.

The tag break-down in the reports is a cool idea, will think about how to add it.

**Comment by gantir** [comment]
> I guess the filter options (`--date`, `--before`, etc.) could be added to `tags` and `report` as well.
> 
> The tag break-down in the reports is a cool idea, will think about how to add it.

**Comment by jotaen** [comment]
@gantir filtering for `tags` and `report` will be included in the upcoming release v1.3
